### ISSUE TITLE
For some reason the YDoc sometimes remains unsynchronized when loading the page

### ISSUE THREAD

**Issue by linspw** [stefanw/channels-yroom#6]
[Gravação de tela de 04-05-2023 20:14:14.webm](https://user-images.githubusercontent.com/43552865/236349603-9b676180-de51-46ca-ad05-bda8555509f3.webm)

This happens on half the tries. It does not generate errors in the console.
Maybe it could be a concurrency issue. I've looked at the snapshot data and it's correct.

My Workaround is force sync again (works with it) 🤔 
```ts
    const ydoc = new Y.Doc()
    this.provider = new HocuspocusProvider({
      url: `ws://localhost:8000/workspace/v1/documents/${this.document.id}`,
      name: String(this.document.id),
      document: ydoc,
      onSynced(data) {
        console.log(data)
      },
    })
    this.provider.on('connect', () => {
      this.provider?.forceSync()
    })

    this.editor = new Editor({
      extensions: [
        StarterKit.configure({
          history: false,
        }),
        Collaboration.configure({
          document: ydoc,
          field: 'content',
        }),
        CollaborationCursor.configure({
          provider: this.provider,
          user: { name: this.getRandomName(), color: this.getRandomColor() },
        }),
      ],
    }
```

Thanks in advance for your attention, Stefan!

Your project is amazing!

**Comment by stefanw** [comment]
I looked into this and I'm currently suspecting that `HocuspocusProvider` cannot read multiple `y-protocol` messages in one message receive (it could and probably should though). I'm looking into a workaround, but it is a bit annoying as I apparently need to split one initial reply (ie sync step 2, sync step 1 and awareness update) into multiple sends which could be one send.

**Comment by stefanw** [comment]
I added a setting to disable message pipelining. And I also found and fixed some bug in regards to syncing.

One other pesky synchronization bug that would lose messages was because of a broken/outdated version of the Python redis package ([see this issue](https://github.com/django/channels_redis/issues/366)). Upgrading to `redis>=4.5.5` (released 8.5.2023) fixes this.

Latest version v0.0.5 fixes these issues.

Docker compose example project includes now a TipTap editor and the [docs have a more detailed guide for Tiptap](https://channels-yroom.readthedocs.io/en/latest/howto/tiptap/).
